pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -51,13 +58,127 @@ func (r *RedisOptionsProvider) GetRedisOptions() (*component.RedisOptions, error
+			component.DefaultBackendRedisStorageURL(),
+		},
+		{
+			&r.options.BackendQueuesURL,
+			component.BackendSecretBackendRedisSecretName,
+			component.BackendSecretBackendRedisQueuesURLFieldName,
+			component.DefaultBackendRedisQueuesURL(),
+		},
+		{
+			&r.options.BackendRedisStorageSentinelHosts,
+			component.BackendSecretBackendRedisSecretName,
+			component.BackendSecretBackendRedisStorageSentinelHostsFieldName,
+			component.DefaultBackendStorageSentinelHosts(),
+		},
+		{
+			&r.options.BackendRedisStorageSentinelRole,
+			component.BackendSecretBackendRedisSecretName,
+			component.BackendSecretBackendRedisStorageSentinelRoleFieldName,
+			component.DefaultBackendStorageSentinelRole(),
+		},
+		{
+			&r.options.BackendRedisQueuesSentinelHosts,
+			component.BackendSecretBackendRedisSecretName,
+			component.BackendSecretBackendRedisQueuesSentinelHostsFieldName,
+			component.DefaultBackendQueuesSentinelHosts(),
+		},
+		{
+			&r.options.BackendRedisQueuesSentinelRole,
+			component.BackendSecretBackendRedisSecretName,
+			component.BackendSecretBackendRedisQueuesSentinelRoleFieldName,
+			component.DefaultBackendQueuesSentinelRole(),
+		},
+		{
+			&r.options.SystemRedisURL,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisURLFieldName,
+			component.DefaultSystemRedisURL(),
+		},
+		{
+			&r.options.SystemRedisMessageBusURL,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisMessageBusRedisURLFieldName,
+			component.DefaultSystemRedisMessageBusURL(),
+		},
+		{
+			&r.options.SystemRedisSentinelsHosts,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisSentinelHosts,
+			component.DefaultSystemRedisSentinelHosts(),
+		},
+		{
+			&r.options.SystemRedisSentinelsRole,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisSentinelRole,
+			component.DefaultSystemRedisSentinelRole(),
+		},
+		{
+			&r.options.SystemMessageBusRedisSentinelsHosts,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisMessageBusSentinelHosts,
+			component.DefaultSystemMessageBusRedisSentinelHosts(),
+		},
+		{
+			&r.options.SystemMessageBusRedisSentinelsRole,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisMessageBusSentinelRole,
+			component.DefaultSystemMessageBusRedisSentinelRole(),
+		},
+		{
+			&r.options.SystemRedisNamespace,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisNamespace,
+			component.DefaultSystemRedisNamespace(),
+		},
+		{
+			&r.options.SystemMessageBusRedisNamespace,
+			component.SystemSecretSystemRedisSecretName,
+			component.SystemSecretSystemRedisMessageBusRedisNamespace,
+			component.DefaultSystemMessageBusRedisNamespace(),
+		},
+	}
+
+	for _, option := range cases {
+		val, err := r.secretSource.FieldValue(option.secretName, option.secretField, option.defValue)
```

Comment:
Same comment as https://github.com/3scale/3scale-operator/pull/486/files#r500173988

Reply:
This is also related to the previous conv about the necessity of reading the secret. 

In this case, we are reconciling the secret and it may not exist (in the beginning). If not existing, default values are read and the secret will be created.

Reply:
You are right.

In this case redis options provider is used only for internal redis so the secret can be created automatically by the operator if it has not been pre-created by the user.